addons/hr_recruitment_survey/models/survey_invite.py

```diff
@@ -1,22 +1,25 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import fields, models, _
+from odoo.tools.misc import clean_context
 
 
 class SurveyInvite(models.TransientModel):
     _inherit = "survey.invite"
 
-    applicant_id = fields.Many2one('hr.applicant', string='Applicant', default=lambda self: self.env.context.get('active_id', None))
+    applicant_id = fields.Many2one('hr.applicant', string='Applicant')
 
     def action_invite(self):
         self.ensure_one()
 
-        if not self.applicant_id.response_id:
-            response = self.applicant_id.survey_id._create_answer(partner=self.applicant_id.partner_id)
-            self.applicant_id.response_id = response.id
-
-        body = _('The survey has been sent to "%s".', self.applicant_id.partner_name)
-        self.applicant_id.message_post(body=body)
+        if self.applicant_id:
+            if not self.applicant_id.response_id:
+                survey = self.applicant_id.survey_id.with_context(clean_context(self.env.context)) 
+                self.applicant_id.write({
+                    'response_id': survey._create_answer(partner=self.applicant_id.partner_id).id
+                })
+            body = _('The survey has been sent to "%s".', self.applicant_id.partner_name)
```

Comment:
Could be great to link records directly in chatter using oe-model, like
```
body = Markup(
    _(<p>'The survey %(survey_link)s has been sent to %(recipient_link)s.</p>',
      survey_link=f'<a href="#" t-att-data-oe-model="{survey._name}" t-att-data-oe-id="{survey.id}">{survey.name}</a>'
      recipient_link=f'<a href="#" t-att-data-oe-model="{self.applicant_id._name}" t-att-data-oe-id="{self.applicant_id.id}">{self.applicant_id.name}</a>'
      )
```

Reply:
Thanks for suggestion, it's indeed way better :)

I was thinking... I think it makes more sense to link to partner_id rather than applicant_id, because we are already on that record. What do you think?
As for Markup, do we really need to escape stuff? :thinking: Not sure

Reply:
Indeed, good idea :)

For Markup actually I think this is not required and we can simply remove the Markup enclosure. Indeed we are giving content for a sanitized field with content we own, so normally not.

Reply:
Hello Hello again @tde-banana-odoo, the solution that you propose here for `body` is way more elegant than how I did it finally, but with this solution I had issues with translation :grimacing: https://runbot.odoo.com/runbot/build/11322587

Do let me know if I could clean it up or if there is something I could do further :)

Reply:
Main issue with that solution is that you are going to extract html to translate (and actually I did the same mistake in my first proposal). If some people translates it wrongly, your html will be broken. Only parts to really translate should be inside the translation marks. We could do something using fstrings (not sure translation marks inside an fstring work, maybe they will have to be extracted) like
```
partner = self.applicant_id.partner_id
body = f"<p>{_('The survey')} <a href="#" data-oe-model="{survey._name}" data-oe-id="{survey.id}">{survey.title}</a> {_('has been sent to')} <a href="#" data-oe-model="{partner._name}" data-oe-id="{partner.id}">{partner.name}</a>."
```

Reply:
Side note: don't forget the final </p> (seems missing)

Reply:
Don't get what you mean by 'final' :sweat_smile: Like a final comment? But I did change `self.applicant_id.survey_id` with `self.survey_id` :thinking:

Reply:
final ``</p>`` (did not see github considered it as html or "i don't know what")